### PR TITLE
Fix/some bugs reported by users

### PR DESCRIPTION
**Pull request by cloudwebrtc**

### CODE REVIEW

File: lib/src/core/engine.dart [hunk 1]
```diff
@@ -823,11 +823,8 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
   }) async {
     final previousAnswer =
         (await subscriber?.pc.getLocalDescription())?.toPBType();
-    final previousOffer =
```
Comment (davidzhao) [inline lib/src/core/engine.dart:826]: can you clarify why this was removed?

File: lib/src/widgets/video_track_renderer.dart [hunk 2]
```diff
@@ -52,10 +53,18 @@ class VideoTrackRenderer extends StatefulWidget {
 
 class _VideoTrackRendererState extends State<VideoTrackRenderer> {
   rtc.RTCVideoRenderer? _renderer;
+  bool _rendererReady = false;
   EventsListener<TrackEvent>? _listener;
   // Used to compute visibility information
   late GlobalKey _internalKey;
 
+  Future<rtc.RTCVideoRenderer?> _initializeRenderer() async {
```
Comment (davidzhao) [inline lib/src/widgets/video_track_renderer.dart:61]: seems like it's not possible for _renderer to be null? would be better to reflect that in the method signature.

File: lib/src/widgets/video_track_renderer.dart [hunk 3]
```diff
@@ -52,10 +53,18 @@ class VideoTrackRenderer extends StatefulWidget {
 
 class _VideoTrackRendererState extends State<VideoTrackRenderer> {
   rtc.RTCVideoRenderer? _renderer;
+  bool _rendererReady = false;
```
Comment (davidzhao) [inline lib/src/widgets/video_track_renderer.dart:56]: if only used for web, could we reflect in variable name? along with a comment it's web-only.

**Review by davidzhao — COMMENTED** [review]
I only understood parts of it. could you clarify/link to the bugs that this PR is fixing?

File: lib/src/core/engine.dart (hunk 1 above)
Comment (cloudwebrtc) [reply lib/src/core/engine.dart:826]: There was an internal exception in flutter-webrtc, so I temporarily removed the code (added from 1.5.5) and added it back after testing
image

**Review by davidzhao — APPROVED** [review]

**Pull request merged**